SQF/dayz_code/compile/dze_buildChecks.sqf

```diff
@@ -12,6 +12,24 @@ if (!canbuild) exitWith {dayz_actionInProgress = false; format[localize "STR_EPO
 if (isNumber (configFile >> "CfgVehicles" >> _classname >> "requireplot")) then {
 	_requireplot = getNumber(configFile >> "CfgVehicles" >> _classname >> "requireplot");
 };
+
+_checkClass = {
+	private ["_checkOK","_distance"];
+
+	_checkOK = false;
+	_distance = DZE_SafeZoneNoBuildDistance;
+
+	{
+		if (typeName _x == "ARRAY") then {
+			if (_x select 0 == _classname) exitWith {_checkOK = true; _distance = _x select 1;};
+		} else {
+			if (_x == _className) exitWith {_checkOK = true};
```

Comment:
Using exitWith here only exits the Then { } else { }; scope, not the count loop, so this loop is always just checking the last element in the array right now.